### PR TITLE
Add tutorial for fzf

### PR DESCRIPTION
**Pull request by xulongwu4**
## Description

Add a documentation for enabling fuzzy autocompletion and key-bindings for fzf.

### Submitter Checklist

- [x] Updated the "lastmod" portion at the top of any modified Markdown files.
- [x] Squashed commits with `git rebase -i` (if needed)

### CODE REVIEW

File: software/fzf/en.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+### Zsh
+``` bash
+FZF_COMPLETION_FILE=/usr/share/zsh/site-functions/_fzf
+[[ -f $FZF_COMPLETION_FILE ]] && source $FZF_COMPLETION_FILE
+```
+
+## Key Bindings
+`fzf` can use specific [key bindings](https://github.com/junegunn/fzf#key-bindings-for-command-line) to trigger a search over a list of files, command history and directories and paste the result onto the command-line.
+To enable these key bindings, add the following lines to the shell configuration file
```
Comment (JoshStrobl) [inline software/fzf/en.md]: Putting on newline not necessary, can probably be a bit more verbose like:

> To enable these keybindings, add the following lines to the shell configuration file depending on the shell you use. This will be `~/.bashrc`, `~/.config/fish/config.fish`, or `~/.zshrc`.

File: software/fzf/en.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+[[ -f $FZF_KEYBINDING_FILE ]] && source $FZF_KEYBINDING_FILE
+```
+
+### Zsh
+``` bash
+FZF_KEYBINDING_FILE=/usr/share/fzf/key-bindings.zsh
+[[ -f $FZF_KEYBINDING_FILE ]] && source $FZF_KEYBINDING_FILE
+```
+
+### Fish
```
Comment (JoshStrobl) [inline software/fzf/en.md]: Let's have the fish section above zsh

File: software/fzf/en.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+``` bash
+FZF_KEYBINDING_FILE=/usr/share/fzf/key-bindings.fish
+[ -f $FZF_KEYBINDING_FILE ] && source $FZF_KEYBINDING_FILE
+```
+
+More tips and examples on the usage of `fzf` can be found on its [github](https://github.com/junegunn/fzf) and [wiki](https://github.com/junegunn/fzf/wiki/examples) pages.
```
Comment (JoshStrobl) [inline software/fzf/en.md]: Lets add this after the "The procedures to enable these features are described below." in Installation (not as a new paragraph however) and maybe rephrase it to:

```
For more tips and examples on the usage of `fzf`, visit its [github](https://github.com/junegunn/fzf) and [wiki](https://github.com/junegunn/fzf/wiki/examples) pages.
```

Comment (xulongwu4) [reply software/fzf/en.md]: I added some explanation about the shell configuration file for this part as well as the `Fuzzy Auto-Completion` part.

File: software/fzf/en.md (hunk 1 above)
Comment (xulongwu4) [reply software/fzf/en.md]: I added some explanation about the shell configuration file for this part as well as the `Fuzzy Auto-Completion` part.

File: software/fzf/en.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
++++
+title = "fzf"
+lastmod = "2019-10-07T15:55:50+02:00"
++++
+# fzf
+[fzf](https://github.com/junegunn/fzf) is a general-purpose command-line fuzzy finder. It can be used as an interactive Unix filter with any list: files, command history, processes, hostnames, bookmarks, git commits, etc.
```
Comment (JoshStrobl) [inline software/fzf/en.md:6]: Needs a new line above it.

File: software/fzf/en.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## Installation 
+
+The fzf project consists of the following components: an `fzf` executable, an `fzf-tmux` script for launching fzf in a tmux pane, shell extensions (including key bindings and command-line fuzzy auto-completion), and a vim/Neovim plugin file. They are all available in the `fzf` package in the Software Center or via eopkg in a terminal:
+``` bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:11]: Needs a new line above it.

File: software/fzf/en.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+sudo eopkg it fzf
+```
+
+The shell extensions are not enabled by default after the `fzf` package is installed. The procedures to enable these features are described below. For more tips and examples on the usage of `fzf`, visit its [github](https://github.com/junegunn/fzf) and [wiki](https://github.com/junegunn/fzf/wiki/examples) pages.
+
+## Fuzzy Auto-Completion
+
+[Fuzzy completion](https://github.com/junegunn/fzf#fuzzy-completion-for-bash-and-zsh) for files and directories can be triggered by a trigger sequence (`**` by default) followed by the `TAB` key for `bash` and `zsh`. To enable this feature, add the following lines to the shell configuration file depending on the shell you use (`~/.bashrc` for `bash` and `~/.zshrc` for `zsh`).
+### Bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:20]: Needs a new line above it.

File: software/fzf/en.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+``` bash
+FZF_COMPLETION_FILE=/usr/share/bash-completion/completions/fzf
+[[ -f $FZF_COMPLETION_FILE ]] && source $FZF_COMPLETION_FILE
+```
+### Zsh
```
Comment (JoshStrobl) [inline software/fzf/en.md:25]: Needs a new line above it.

File: software/fzf/en.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+``` bash
+FZF_COMPLETION_FILE=/usr/share/zsh/site-functions/_fzf
+[[ -f $FZF_COMPLETION_FILE ]] && source $FZF_COMPLETION_FILE
+```
+
+## Key Bindings
+
+`fzf` can use specific [key bindings](https://github.com/junegunn/fzf#key-bindings-for-command-line) to trigger a search over a list of files, command history and directories and paste the result onto the command-line. Follow these steps to set up the key bindings for your favorite shell.
+### Bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:34]: Needs a new line above it.

File: software/fzf/en.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+Add the following content to `~/.bashrc`
+``` bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:36]: Needs a new line above it.

File: software/fzf/en.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+sudo eopkg it fzf
+```
+
+The shell extensions are not enabled by default after the `fzf` package is installed. The procedures to enable these features are described below. For more tips and examples on the usage of `fzf`, visit its [github](https://github.com/junegunn/fzf) and [wiki](https://github.com/junegunn/fzf/wiki/examples) pages.
+
+## Fuzzy Auto-Completion
+
+[Fuzzy completion](https://github.com/junegunn/fzf#fuzzy-completion-for-bash-and-zsh) for files and directories can be triggered by a trigger sequence (`**` by default) followed by the `TAB` key for `bash` and `zsh`. To enable this feature, add the following lines to the shell configuration file depending on the shell you use (`~/.bashrc` for `bash` and `~/.zshrc` for `zsh`).
+### Bash
+``` bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:21]: Needs a new line above it.

File: software/fzf/en.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+## Key Bindings
+
+`fzf` can use specific [key bindings](https://github.com/junegunn/fzf#key-bindings-for-command-line) to trigger a search over a list of files, command history and directories and paste the result onto the command-line. Follow these steps to set up the key bindings for your favorite shell.
+### Bash
+Add the following content to `~/.bashrc`
+``` bash
+FZF_KEYBINDING_FILE=/usr/share/fzf/key-bindings.bash
+[[ -f $FZF_KEYBINDING_FILE ]] && source $FZF_KEYBINDING_FILE
+```
+### Fish
```
Comment (JoshStrobl) [inline software/fzf/en.md:40]: Needs a new line above it.

File: software/fzf/en.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+First create the following directory if it does not already exist
+```bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:42]: Needs a new line above it.

File: software/fzf/en.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+mkdir -p $HOME/.config/fish/functions
+```
+`cd` into this directory and make a file `fish_user_key_bindings.fish` with the following content
```
Comment (JoshStrobl) [inline software/fzf/en.md:45]: Needs a new line above it and add a `:` (colon) after "content"

File: software/fzf/en.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+``` bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:46]: Needs a new line above it.

File: software/fzf/en.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+function fish_user_key_bindings
+  fzf_key_bindings
+end
+```
+Then create the following symlink
```
Comment (JoshStrobl) [inline software/fzf/en.md:51]: Needs a new line above it. Also add a `:` (colon) after the word "symlink"

File: software/fzf/en.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+``` bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:52]: Needs a new line above it.

File: software/fzf/en.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+ln -s /usr/share/fzf/key-bindings.fish $HOME/.config/fish/functions/fzf_key_bindings.fish
+```
+### Zsh
```
Comment (JoshStrobl) [inline software/fzf/en.md:55]: Needs a new line above it.

File: software/fzf/en.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+Add the following content to `~/.zshrc`
+``` bash
```
Comment (JoshStrobl) [inline software/fzf/en.md:57]: Needs a new line above it.

File: software/fzf/en.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+`fzf` can use specific [key bindings](https://github.com/junegunn/fzf#key-bindings-for-command-line) to trigger a search over a list of files, command history and directories and paste the result onto the command-line. Follow these steps to set up the key bindings for your favorite shell.
+### Bash
+Add the following content to `~/.bashrc`
+``` bash
+FZF_KEYBINDING_FILE=/usr/share/fzf/key-bindings.bash
+[[ -f $FZF_KEYBINDING_FILE ]] && source $FZF_KEYBINDING_FILE
+```
+### Fish
+First create the following directory if it does not already exist
```
Comment (JoshStrobl) [inline software/fzf/en.md:41]: Add a colon after "exist"

**Comment by xulongwu4** [comment]
Comments are resolved.
